airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/components.py

```diff
@@ -42,15 +46,25 @@ def get_headers(self):
     @backoff.on_exception(
         backoff.expo,
         DefaultBackoffException,
+        max_tries=2,
         on_backoff=lambda details: logger.info(
             f"Caught retryable error after {details['tries']} tries. Waiting {details['wait']} seconds then retrying..."
         ),
         max_time=300,
     )
     def _get_refresh_access_token_response(self):
         try:
+            request_url = self.get_token_refresh_endpoint()
+            request_headers = self.get_headers()
+            request_body = self.build_refresh_request_body()
+
+            logger.info(f"Sending request to URL: {request_url}")
+            
             response = requests.request(
-                method="POST", url=self.get_token_refresh_endpoint(), data=self.build_refresh_request_body(), headers=self.get_headers()
+                method="POST"
+                , url=request_url
```

Comment:
Is this request body a new CDK requirement? Is that why we need the custom component?

Reply:
This was existing code, I just separated the code and gave it a bit of format. I also added max_retries to the backoff extension. Since this uses a client_id and client_secret, this customAuthenticator is needed to do the proper bearer token pass?